R/plumber.R

```diff
@@ -254,44 +255,23 @@ plumber <- R6Class(
     #' @param port a number or integer that indicates the server port that should
     #' be listened on. Note that on most Unix-like systems including Linux and
     #' Mac OS X, port numbers smaller than 1025 require root privileges.
-    #' @param swagger a function that enhances the existing OpenAPI Specification.
     #' @param debug `TRUE` provides more insight into your API errors.
-    #' @param swaggerCallback a callback function for taking action on the url for swagger page.
+    #' @param callback a callback function for taking action on UI url.
     #' @details
     #' `port` does not need to be explicitly assigned.
     #'
-    #' `swagger` should be either a logial or a function . When `TRUE` or a
-    #' function, multiple handles will be added to `plumber` object. OpenAPI json
-    #' file will be served on paths `/openapi.json` and `/swagger.json`. Swagger UI
-    #' will be served on paths `/__swagger__/index.html` and `/__swagger__/`. When
-    #' using a function, it will receive the plumber router as the first parameter
-    #' and current OpenAPI Specification as the second. This function should return a
-    #' list containing OpenAPI Specification.
-    #' See \url{http://spec.openapis.org/oas/v3.0.3}
-    #'
-    #' `swaggerCallback` When set, it will be called with a character string corresponding
-    #' to the swagger UI url. It allows RStudio to open swagger UI when plumber router
-    #' run method is executed using default `plumber.swagger.url` option.
-    #' @examples
-    #' \dontrun{
-    #' pr <- plumber$new
-    #' swagger <- function(pr_, spec) {
-    #'   spec$servers[[1]]$description <- "MyCustomAPISpec"
-    #'   spec
-    #' }
-    #' pr$run(swagger = swagger)
-    #' }
+    #' `callback` When set, it will be called with a character string corresponding
+    #' to the UI url. It allows RStudio to open UI when plumber router
+    #' run method is executed using default `plumber.ui.url` option.
     run = function(
       host = '127.0.0.1',
       port = getOption('plumber.port'),
-      swagger = interactive(),
       debug = interactive(),
```

Comment:
Add back `swagger` argument (so as not to break existing to code), then throw a deprecation warning. If it's somewhat straight-forward/sensible, also try translating the value to the new approach